tests/Console/ConfigurationResolverTest.php

```diff
@@ -943,6 +948,140 @@ public function testResolveRulesWithOption()
         );
     }
 
+    /**
+     * @param array $expected
+     * @param array $rules
+     *
+     * @dataProvider provideResolveRulesCases
+     */
+    public function testResolveRules(array $expected, array $rules)
+    {
+        $ruleSet = $this->createRuleSetToTestWith($rules);
+
+        $this->assertSameRules($expected, $ruleSet->getRules());
```

Comment:
moved :)
as a bonus, with the new added logic for testing the resolving of sets, more logic of the RuleSet class can be tested (so no logic changes, just more test coverage)